test/module/irohad/torii/torii_transport_command_test.cpp

```diff
@@ -0,0 +1,301 @@
+ * @when calling ListTorii
+ * @then ensure that CommandService haven't called handleTransactionBatch
+ *       and StatusBus update status tx num times
+ */
+TEST_F(ToriiTransportCommandTest, ListToriiInvalid) {
+  grpc::ServerContext context;
+  google::protobuf::Empty response;
+
+  iroha::protocol::TxList request;
+  for (size_t i = 0; i < kTimes; ++i) {
+    request.add_transactions();
+  }
+
+  shared_model::validation::Answer error;
+  error.addReason(std::make_pair("some error", std::vector<std::string>{}));
+  EXPECT_CALL(*tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Return(error));
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(0);
+  EXPECT_CALL(*status_bus, publish(_)).Times(kTimes);
+
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service
+ *        and some number of valid transactions
+ *        and one stateless invalid tx
+ * @when calling ListTorii
+ * @then ensure that CommandService haven't called handleTransactionBatch
+ *       and StatusBus publishes statelessInvalid for all txes
+ */
+TEST_F(ToriiTransportCommandTest, ListToriiPartialInvalid) {
+  grpc::ServerContext context;
+  google::protobuf::Empty response;
+
+  iroha::protocol::TxList request{};
+  for (size_t i = 0; i < kTimes; ++i) {
+    request.add_transactions();
+  }
+
+  int counter = 0;
+  EXPECT_CALL(*tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Invoke([&counter](const auto &) mutable {
+        shared_model::validation::Answer res;
+        if (counter++ == kTimes - 1) {
+          res.addReason(
+              std::make_pair("some error", std::vector<std::string>{}));
+        }
+        return res;
+      }));
+  EXPECT_CALL(
+      *batch_factory,
+      createTransactionBatch(
+          A<const shared_model::interface::types::SharedTxsCollectionType &>()))
+      .Times(kTimes - 1);
+
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(kTimes - 1);
+  EXPECT_CALL(*status_bus, publish(_)).WillOnce(Invoke([](auto status) {
+    EXPECT_FALSE(status->statelessErrorOrCommandName().empty());
+  }));
+
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service and command_service with empty status stream
+ * @when calling StatusStream on transport
+ * @then Ok status is eventually returned without any fault
+ */
+TEST_F(ToriiTransportCommandTest, StatusStreamEmpty) {
+  grpc::ServerContext context;
+  iroha::protocol::TxStatusRequest request;
+
+  EXPECT_CALL(*command_service, getStatusStream(_))
+      .WillOnce(Return(rxcpp::observable<>::empty<std::shared_ptr<
+                           shared_model::interface::TransactionResponse>>()));
+
+  ASSERT_TRUE(transport_grpc->StatusStream(&context, &request, nullptr).ok());
+}
+
+/**
+ * @given torii service and number of invalid transactions
+ * @when calling StatusStream
+ * @then ServerWriter call Write method the same number of times
```

Comment:
This test covers the same logic kTimes times without assertions on valid transformations done with stream. It can be split into several test cases:
- initial timeout is used when NOT_RECEIVED status is written into stream
- nonfinal timeout is used when other status is written
- nothing is written after ServerContext returns true on `IsCancelled` call

Reply:
The last one is really dangerous to use, since there's a `reinterpret_cast` on incompatible types. I wouldn't consider it as a good idea

Reply:
OK, please create an issue on JIRA for `MockServerWriter` so that it can be used without `reinterpret_cast`.
Since there is no possibility to verify timeout value at the moment, another task should be created as stated before.
This test case is the same as `StatusStreamOnNotRecieved` right now, it looks like it can be removed.

Reply:
IR-132